ting-bill-split.go

```diff
@@ -9,8 +9,30 @@ import (
 	"log"
 	"os"
 	"strconv"
+
+	"github.com/BurntSushi/toml"
 )
 
+type bill struct {
+	Minutes   float64 `toml:"minutes"`
+	Messages  float64 `toml:"messages"`
+	Megabytes float64 `toml:"megabytes"`
+	Devices   float64 `toml:"devices"`
+	Extras    float64 `toml:"extras"`
+	Fees      float64 `toml:"fees"`
+}
+
+func parseBill(r io.Reader) (bill, error) {
+	var b bill
+	_, err := toml.DecodeReader(r, &b)
+
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Don't fatal in these functions. Only `main` should exit the program. Generally people will pass errors up the stack to let someone else deal with them. In this case dealing with the error might just mean exiting the program but it's not `parseBill`'s place to make that call so pass it along.

You can rewrite this function as
```
func parseBill(r io.Reader) (bill, error) {
	var b bill
	if _, err := toml.DecodeReader(r, &b); err != nil {
		return bill{}, err
	}
	return b, nil
}
```

Which would return a zero-value `bill` in the error case and a `nil` error in the positive case.

OR you could reasonably just do 
```
func parseBill(r io.Reader) (bill, error) {
	var b bill
	_, err := toml.DecodeReader(r, &b)
	return b, err
}
```

The only real difference is philosophical. In this method you may be returning an incomplete or corrupt `bill` if there was an error BUT THEN AGAIN some people will say that doesn't matter because it's the caller's responsibility ignore the `bill` if the error is set.